File: yumi/sdk/csharp/ToolParameter.cs
namespace Yumi;

/// <summary>
/// Describes a single parameter in a tool's schema.
/// </summary>
public sealed class ToolParameter
{
    public string Name { get; }
    public string Type { get; }
    public string Description { get; }
    public bool Required { get; }

    public ToolParameter(string name, string type, string description, bool required = true)
    {
        Name = name;
        Type = type;
        Description = description;
        Required = required;
    }
}

File: kumi/sdk/java/src/main/java/io/kumi/ToolParameter.java
package io.kumi;

/**
 * Describes a single parameter in a tool's schema.
 */
public class ToolParameter {
    private final String name;
    private final String type;
    private final String description;
    private final boolean required;

    public ToolParameter(String name, String type, String description) {
        this(name, type, description, true);
    }

    public ToolParameter(String name, String type, String description, boolean required) {
        this.name = name;
        this.type = type;
        this.description = description;
        this.required = required;
    }

    public String getName() { return name; }
    public String getType() { return type; }
    public String getDescription() { return description; }
    public boolean isRequired() { return required; }
}
